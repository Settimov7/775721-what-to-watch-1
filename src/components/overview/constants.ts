type Level = {
  min: number;
  max?: number;
  value: string;
}

export const LEVELS: Level[] = [
  {
    min: 0,
    max: 3,
    value: `Bad`,
  },
  {
    min: 3,
    max: 5,
    value: `Normal`,
  },
  {
    min: 5,
    max: 8,
    value: `Good`,
  },
  {
    min: 8,
    max: 10,
    value: `Very good`,
  },
  {
    min: 10,
    value: `Awesome`,
  },
];
