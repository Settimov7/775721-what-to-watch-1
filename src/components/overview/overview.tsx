import * as React from 'react';

interface Props {
  rating: number,
  director: string,
  description: string,
  starring: string[],
}

export const Overview: React.FunctionComponent<Props> = (props) => {
  const {rating, director, description, starring} = props;

  return(
    <React.Fragment>
      <div className="movie-rating">
        <div className="movie-rating__score">{rating}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">Very good</span>
          <span className="movie-rating__count">240 ratings</span>
        </p>
      </div>

      <div className="movie-card__text">
        <p>{description}</p>

        <p className="movie-card__director">
          <strong>Director: {director}</strong>
        </p>

        <p className="movie-card__starring">
          <strong>
            Starring: {starring.reduce((result, actor, index) => {
              const isLast = index === starring.length - 1;

              return isLast ? `${result} ${actor}` : `${result} ${actor}, `;
          }, ``)} and other
          </strong>
        </p>
      </div>
    </React.Fragment>
  );
};
