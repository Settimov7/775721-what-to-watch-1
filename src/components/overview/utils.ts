import {LEVELS} from './constants';

export const getLevelByRating = (rating: number): string =>
  LEVELS.find(({min, max}) => (rating>= min && rating < max) || rating === min).value;
