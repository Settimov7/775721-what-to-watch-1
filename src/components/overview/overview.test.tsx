import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Overview} from './overview';

it(`Overview correctly renders`, () => {
  const props = {
    rating: 5,
    director: `director`,
    description: `description`,
    starring: [`person-1`, `person-2`, `person-3`],
  };
  const overview = renderer.create(
    <Overview {...props}/>
  ).toJSON();

  expect(overview).toMatchSnapshot();
});
