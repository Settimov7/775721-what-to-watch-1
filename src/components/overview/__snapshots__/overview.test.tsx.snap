// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview correctly renders 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        240 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
         person-1,  person-2,  person-3
         and other
      </strong>
    </p>
  </div>,
]
`;
